cmd/tink-worker/internal/registry.go

```diff
@@ -0,0 +1,75 @@
+package internal
+
+import (
+	"context"
+	"encoding/base64"
+	"encoding/json"
+	"io"
+	"os"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var log *logrus.Logger
+
+type RegistryConnDetails struct {
+	registry,
+	user,
+	pwd string
+	logger *logrus.Logger
+}
+
+func SetLogger(logger *logrus.Logger) {
+	log = logger
+}
+
+func NewRegistryConnDetails(registry, user, pwd string, logger *logrus.Logger) *RegistryConnDetails {
+	return &RegistryConnDetails{
+		registry: registry,
+		user:     user,
+		pwd:      pwd,
+		logger:   logger,
+	}
+}
+
+func (r *RegistryConnDetails) NewClient() (*client.Client, error) {
+	if r.registry == "" {
+		return nil, errors.New("required DOCKER_REGISTRY")
+	}
+	c, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
```

Comment:
This will likely need to be updated to work with non-env based registry url, I think potentially just adding client.WithHost(r.registry) would be sufficient

Reply:
That sounds about right and good.

Is `client.WithAPIVersionNegotiation` a default option? I'll look into it.

This `client.FromEnv` options resembles the Tink NewClient example I offered up in feedback elsewhere, it seems I was inspired.

Reply:
I'm going with `c, err := client.NewClientWithOpts(client.FromEnv, client.WithHost(r.registry), client.WithAPIVersionNegotiation())`. I suspect that including `client.FromEnv` may be needed for some other DOCKER_ variables that we are not exposing elsewhere.

```
Supported environment variables:
DOCKER_HOST to set the url to the docker server.
DOCKER_API_VERSION to set the version of the API to reach, leave empty for latest.
DOCKER_CERT_PATH to load the TLS certificates from.
DOCKER_TLS_VERIFY to enable or disable TLS verification, off by default.
```

(Seems we may have invented DOCKER_REGISTRY for what DOCKER_HOST was already providing?)

Reply:
There is going to be a slight distinction between DOCKER_HOST and our use of DOCKER_REGISTRY, in this case DOCKER_HOST is referring to the docker daemon, which we would likely want to generally *not* override here in favor of the local docker daemon